proposals/0000-package-dependency-import-syntax.md

```diff
@@ -0,0 +1,155 @@
+
+```swift
+@package(url: "https://github.com/jpsim/Yams.git", from: "2.0.0")
+import Yams
+```
+
+This script can be executable by either `swift MyScript.swift` or `swift run MyScript.swift` which will utilize Swift package managers dependency fetching and resolution logic.
+
+The `@package(...)`syntax will allow the same parameters as SwiftPM's [Package Dependency Description](https://docs.swift.org/package-manager/PackageDescription/PackageDescription.html#package-dependency) which should feel natural to developers familiar with Swift Package Manager manifest syntax.  
+
+## Detailed design
+
+
+#### Basic usage
+The most common case is expected to be importing a dependency where the product name is equivalent to the imported module name.
+```swift
+@package(url: "https://github.com/jpsim/Yams.git", from: "2.0.0")
+import Yams
+```
+
+When a dependency product and target names do not match, the product must be explicitly specified.
+
+For an example, lets look at the Package description below for `swift-tools-support-core`. 
+```swift
+let package = Package(
+    name: "swift-tools-support-core",
+    products: [
+        .library(
+            name: "SwiftToolsSupport",
+            type: .dynamic,
+            targets: ["TSCBasic", "TSCUtility"]),
+        .library(
+            name: "SwiftToolsSupport-auto",
+            targets: ["TSCBasic", "TSCUtility"]),
+
+        .library(
+            name: "TSCTestSupport",
+            targets: ["TSCTestSupport"]),
+        .executable(
+            name: "TSCTestSupportExecutable",
+            targets: ["TSCTestSupportExecutable"]),
+    ],
+    // Rest of Package.swift truncated for brevity .....
+}
+```
+
+With the above Package descriptions, the `SwiftToolsSupport` product does not match a target within the package description it must be specified explicitly.
+
+```swift
+@package(url: "https://github.com/apple/swift-tools-support-core.git", products: ["SwiftToolsSupport"]) 
+import SwiftToolsSupport 
+```
+
+#### Dependency version requirements
+
+The `@package(..)` annotation would also allow a developer to specify requirements to ensure your scripts are reproducible. The syntax will contain the same level of control and specificity as `PackageDependencyDescription.Requirement` which should feel like a natural to developers familiar with Swift Package Manager manifest syntax.  
+
+*Revision*
+```swift
+@package(url: "https://github.com/jpsim/Yams.git", .revision( "065675b3d1364a6f63b94a9c89be2e9ed0a4c3a1")) 
+import Yams 
+```
+
+*Branch*
+```swift
+@package(url: "https://github.com/jpsim/Yams.git", .branch("releases/1.0")) 
+import Yams 
+```
+
+*Exact Version*
+```swift
+@package(url: "https://github.com/jpsim/Yams.git", .exact("1.0")) 
+import Yams 
+```
+
+*Range of version*
+```swift
+@package(url: "https://github.com/jpsim/Yams.git", .from("1.0")) 
```

Comment:
same comment as above, this should say 1.0.0 instead of 1.0

Reply:
Actually, this should be `@package(url: "https://github.com/jpsim/Yams.git", from: "1.0.0")

Reply:
Instead of giving all those examples separately, shouldn't we instead state that we support the same API as SwiftPM dependencies, and show just one snippet with a 2-3 examples together?